server/routes/userRoutes.js

```diff
@@ -106,18 +93,63 @@ module.exports = (db) => {
     const { host_name } = req.body.params;
     dbHelper
       .getWebsiteIDByHostname(host_name)
-      // .getWebsiteIDByHostname("reddit.com")
-      .then((site) => dbHelper.addWebsiteToBlacklist(userId, site.id))
-      .catch((err) => res.json(err));
+      .then((site) => {
+        if (!site) {
+          // Removes '.com' from end of hostname for DB.
+          const remSuffix = host_name.split(".")[0];
+          // Capitalizes first letter of hostname
+          const name = remSuffix.charAt(0).toUpperCase() + remSuffix.slice(1);
+          const category = null;
+          // Creating the website in the database before it can be added to their blacklist
+          dbHelper
+          .addWebsite(host_name, name, category)
+          .then((site) => {
+            dbHelper
+              .addWebsiteToBlacklist(userId, site.id)
+              .then((blacklist) => res.status(201).json(blacklist))
+              .catch((err) => res.status(500).json(err));
+          });
+        } else {
+          dbHelper
+            .addWebsiteToBlacklist(userId, site.id)
```

Comment:
One last thing... a `.catch` is not provided for line 104 so if `.addWebsite` fails on line 105, the error will be uncaught. Here you can change promise nesting to promise chaining by prefixing `return` to line 107, and then moving lines 109 and 110 to the end of line 111 (remove the semicolon). Now the `.catch` on line 110 will catch both for both `.addWebsite` and `.addWebsiteToBlacklist`

Reply:
Good catch, pun intended 😉